### PR TITLE
Validate and correct repo url for discord button

### LINKED ISSUE #33: Add support for ssh remote urls
When editing a file in a repository with a ssh remote url like `git@github.com:owner/repo`, this plugin fails to add the view repository button, because it isn't a valid uri:
`[presence.nvim] Failed to set activity in Discord: Received error event - child "activity" fails because [child "buttons" fails because ["buttons" at position 0 fails because [child "url" fails because ["url" must be a valid uri]]]] (code 4000)`

As a result no rich presence status is shown at all.

### PR DESCRIPTION
**Pull request by Saecki**
Fixes #33

### CODE REVIEW

**Review by andweeb — CHANGES_REQUESTED** [review]
Thanks for the PR @Saecki! It totally slipped my mind that the git remote URL could potentially be SSH URLs (and include other protocols/formats), and didn't realize Discord would error out on invalid URLs.

It mostly looks good to go! Just got a minor question and some code quality feedback. :) I'd really like to get this in since unfortunately the Discord URI validation disables updating the Rich Presence activity.

File: lua/presence/init.lua [hunk 1]
```diff
@@ -693,6 +704,37 @@ function Presence:get_buttons(buffer, parent_dirpath)
...
+        -- Check if repo url uses a valid protocol
+        local protocols = {
+            "ftp",
+            "git",
+            "http",
+            "https",
+            "ssh",
+        }
+        local protocol, relative = repo_url:match("^(.+)://(.+)$")
+        if contains(protocols, protocol) == false or relative == nil then
```
[the remark this answers is not in the document]
Comment (Saecki) [reply lua/presence/init.lua:724]: Yea this must look stupid, thanks for the feedback, I'm pretty new to lua

File: lua/presence/init.lua [hunk 2]
```diff
@@ -693,6 +704,37 @@ function Presence:get_buttons(buffer, parent_dirpath)
...
+            self.log:debug(string.format("Repository URL uses invalid protocol: %s", repo_url))
+            return nil
+        end
+
+        -- Check if repo url uses has the user specified
```
[the remark this answers is not in the document]
Comment (Saecki) [reply lua/presence/init.lua:729]: Yup

File: lua/presence/init.lua [hunk 3]
```diff
@@ -693,6 +704,37 @@ function Presence:get_buttons(buffer, parent_dirpath)
...
+        local user, path = relative:match("^(.+)@(.+)$")
+        if user and path then
```
[the remark this answers is not in the document]
Comment (Saecki) [reply lua/presence/init.lua:731]: You can specify a username like this: `https://saecki@github.com/saecki/presence.nvim`. That will prompt the use to log in with that username if opened in the browser. And Firefox will warn you.

[the remark this answers is not in the document]
Comment (andweeb) [reply lua/presence/init.lua:720]: Ooh interesting, TIL! In that case I'll resolve this 👍

File: lua/presence/init.lua [hunk 4]
```diff
@@ -721,12 +710,12 @@ function Presence:get_buttons(buffer, parent_dirpath)
             "ssh",
         }
         local protocol, relative = repo_url:match("^(.+)://(.+)$")
-        if contains(protocols, protocol) == false or relative == nil then
+        if not vim.tbl.contains(protocols, protocol) or not relative then
```
Comment (andweeb) [inline lua/presence/init.lua:713]: Oop, you're close but not quite 😅 The function name is `tbl_contains` (not a field on `vim.tbl` which doesn't exist):
> `Error executing vim.schedule lua callback: ...s/andrew/Code/presence.nvim/lua/presence/init.lua:7
13: attempt to index field 'tbl' (a nil value)`

Comment (Saecki) [reply lua/presence/init.lua]: Oops

**Review by andweeb — APPROVED** [review]
Awesome, good work!  Thanks again for the PR @Saecki :)

**Pull request merged**